plugins/generator-1.16.5/forge-1.16.5/templates/item.java.ftl

```diff
@@ -329,6 +348,38 @@ import net.minecraft.entity.ai.attributes.Attributes;
         }
         </#if>
 
+        <#if data.isFood && (hasProcedure(data.onFinishUsingItem) || (data.resultItem?? && !data.resultItem.isEmpty()))>

```

Comment:
This procedure should also work when it is not food, onFinishUsingItem UI element is present for all items